src/service/resources-impl.js

```diff
@@ -506,7 +508,8 @@ export class Resources {
     const parentResource = Resource.forElement(parentElement);
     subElements = elements_(subElements);
 
-    this.discoverResourcesForArray_(parentResource, subElements, resource => {
+    this.discoverResourcesForArray_(parentResource, subElements,
+    /* deepSubtree */ true, resource => {
```

Comment:
To be symmetric with `layout`, yes you are correct but I think the difference is that laying out children requires custom component logic where as unlayout doesn't. I am open to not doing this for `unlayout` but I think doing it is the safer approach to eliminate bugs like #4503.